### ISSUE TITLE
MultiViewSubviewVisitor output varies between projects

### ISSUE THREAD

**Issue by robmaceachern** [nathantannar4/Engine#15]
I'm seeing some strange behavior when using `MultiViewSubviewVisitor` in one of my projects. Here are a couple of test cases that fail (they succeed on a fresh project).

```swift
import XCTest
import Engine
import SwiftUI


final class EngineTests: XCTestCase {

    func testMultiViewSubviewVisitor_group() {
        let content = Group {
            Text("One")
            Text("Two")
            Text("Three")
        }

        var visitor = MultiViewSubviewVisitor()
        content.visit(visitor: &visitor)
        // XCTAssertEqual failed: ("1") is not equal to ("3")
        XCTAssertEqual(visitor.subviews.count, 3)
        print(visitor.subviews)
    }

    func testMultiViewSubviewVisitor_section() {
        let content = Section {
            Text("One")
            Text("Two")
            Text("Three")
        } header: {
            Text("Header")
        } footer: {
            Text("Footer")
        }

        var visitor = MultiViewSubviewVisitor()
        content.visit(visitor: &visitor)
        // XCTAssertEqual failed: ("3") is not equal to ("5")
        XCTAssertEqual(visitor.subviews.count, 5)
        print(visitor.subviews)
    }
}
```

Using Xcode 16.1 and [Tuist](https://github.com/tuist/tuist) for project generation.

~Unfortunately the failing project is closed source and I haven't been able to repro on a sample project yet.~ Sample project can be found here: https://github.com/robmaceachern/MultiViewIssue

In my debugging it looks like the difference is that the [`MultiViewProtocolDescriptor.conformance(of: Self.self)`](https://github.com/nathantannar4/Engine/blob/main/Sources/EngineCore/MultiViewVisitor.swift#L149) check is failing in the bad project.

<details>

<img width="2037" alt="image" src="https://github.com/user-attachments/assets/574cd0c4-5dd5-4eff-96c1-868e47a73154">

</details>

Any idea what could be influencing this different behavior?

**Comment by robmaceachern** [comment]
I've created a minimal sample project here: https://github.com/robmaceachern/MultiViewIssue

The test file is [here](https://github.com/robmaceachern/MultiViewIssue/blob/main/MultiViewIssue/Tests/MultiViewIssueTests.swift).

**Comment by nathantannar4** [comment]
Fixed in `1.9.0`
